api/git/git.go

```diff
@@ -14,39 +14,40 @@ import (
 )
 
 // Service represents a service for managing Git.
-type Service struct{}
+type Service struct {
+	httpsCli *http.Client
+}
 
 // NewService initializes a new service.
 func NewService(dataStorePath string) (*Service, error) {
-	service := &Service{}
+	service := &Service{
+		&http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+			Timeout: 120 * time.Second,
+		},
+	}
 
 	return service, nil
 }
 
 // ClonePublicRepository clones a public git repository using the specified URL in the specified
 // destination folder.
 func (service *Service) ClonePublicRepository(repositoryURL, referenceName string, destination string) error {
-	return cloneRepository(repositoryURL, referenceName, destination)
+	return cloneRepository(repositoryURL, referenceName, destination, service.httpsCli)
 }
 
 // ClonePrivateRepositoryWithBasicAuth clones a private git repository using the specified URL in the specified
 // destination folder. It will use the specified username and password for basic HTTP authentication.
 func (service *Service) ClonePrivateRepositoryWithBasicAuth(repositoryURL, referenceName string, destination, username, password string) error {
 	credentials := username + ":" + url.PathEscape(password)
 	repositoryURL = strings.Replace(repositoryURL, "://", "://"+credentials+"@", 1)
-	return cloneRepository(repositoryURL, referenceName, destination)
+	return cloneRepository(repositoryURL, referenceName, destination, service.httpsCli)
 }
 
-func cloneRepository(repositoryURL, referenceName string, destination string) error {
-	customClient := &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		},
-
-		Timeout: 15 * time.Second,
-	}
-
-	client.InstallProtocol("https", githttp.NewClient(customClient))
+func cloneRepository(repositoryURL string, referenceName string, destination string, httpsCli *http.Client) error {
+	client.InstallProtocol("https", githttp.NewClient(httpsCli))
```

Comment:
what about installing the client in the `NewService` method as well? Instead of having to pass an extra parameter?